Applications/WebPortal/WebApp/library/source/components/single_security_input.tsx

```diff
@@ -0,0 +1,690 @@
+import { css, StyleSheet } from 'aphrodite';
+import * as Nexus from 'nexus';
+import * as React from 'react';
+import { Button, DisplaySize, HLine, Modal } from '..';
+
+interface Properties {
+
+  /** The size at which the component should be displayed at. */
+  displaySize: DisplaySize;
+
+  /** Determines if the component is readonly. */
+  readonly?: boolean;
+
+  /** The list of securities to display. */
+  value?: Nexus.Security;
+
+  /** Called when the list of values changes.
+   * @param value - The new security
+   */
+  onChange?: (value: Nexus.Security) => void;
+}
+
+interface State {
+  inputString: string;
+  isEditing: boolean;
+  localValue: Nexus.Security;
+  isSelected: boolean;
+}
+
+/** A component that displays a list of securities. */
+export class SingleSecurityInput extends React.Component<Properties, State> {
+  public static readonly defaultProps = {
+    value: '',
+    readonly: false,
+    onChange: () => {}
+  }
+
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      inputString: '',
+      isEditing: false,
+      localValue: this.props.value,
+      isSelected: false
+    }
+    this.addEntry = this.addEntry.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onSubmitChange = this.onSubmitChange.bind(this);
+    this.removeEntry = this.removeEntry.bind(this);
+    this.selectEntry = this.selectEntry.bind(this);
+    this.onClose = this.onClose.bind(this);
+    this.onOpen = this.onOpen.bind(this);
+  }
+
+  public render() {
+    const visibility = (() => {
+      if(!this.state.isEditing) {
+        return SingleSecurityInput.STYLE.hidden;
+      } else {
+        return null;
+      }
+    })();
+    const inputField = (() => {
```

Comment:
No need for this to be within a function or even declared here. Inline this directly into the JSX below.